assets/js/modules/tagmanager/setup.js

```diff
@@ -40,38 +40,46 @@ import { addFilter, removeFilter } from '@wordpress/hooks';
 /**
  * Internal dependencies
  */
-import { isValidAccountID, isValidContainerID } from './util';
+import {
+	getContainers,
+	isValidAccountID,
+	isValidContainerID,
+} from './util';
 
 const ACCOUNT_CREATE = 'account_create';
 const CONTAINER_CREATE = 'container_create';
+const USAGE_CONTEXT_WEB = 'web';
+const USAGE_CONTEXT_AMP = 'amp';
 
 class TagmanagerSetup extends Component {
 	constructor( props ) {
 		super( props );
 
+		const { ampEnabled, ampMode } = global.googlesitekit.admin;
 		const { settings } = global.googlesitekit.modules.tagmanager;
-		const usageContext = global.googlesitekit.admin.ampMode === 'primary' ? 'amp' : 'web';
-		const containerKey = usageContext === 'amp' ? 'ampContainerID' : 'containerID';
+		const ampUsageContext = ampMode === 'primary' ? USAGE_CONTEXT_AMP : [ USAGE_CONTEXT_WEB, USAGE_CONTEXT_AMP ];
 
 		this.state = {
+			ampEnabled,
+			ampMode,
 			isLoading: true,
 			accounts: [],
-			containers: [],
+			containersWeb: [],
```

Comment:
I don't think we need to do this renaming. "Web" is the standard here. "AMP" needs to be flagged, but web is overly verbose. I get it for the distinction, but it's different from how we do it on the server. And in a way it clutters this PR quite a bit too :)

Let's leave this as `containers`.

Reply:
I think the naming change makes it easier to understand by removing the need to infer whether `containers` refers to web containers or a collection that could potentially contain both. I understand this adds a few lines to the diff, but I think they're worth it.

There are some cases where this naming separation needs to remain to avoid making things more complicated by reassigning variables, etc; (e.g. `requestTagManagerAccountsContainers`) I assume those are okay to keep?